jdbc/src/java/org/apache/hive/jdbc/CsrfHttpRequestInterceptor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hive.jdbc;
+
+import org.apache.http.HttpException;
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.protocol.HttpContext;
+
+import java.io.IOException;
+
+public class CsrfHttpRequestInterceptor implements HttpRequestInterceptor {
+
+  // Note : This implements HttpRequestInterceptor rather than extending
+  // HttpRequestInterceptorBase, because that class is an auth-specific
+  // class and refactoring would kludge too many things that are potentially
+  // public api.
+  //
+  // At the base, though, what we do is a very simple thing to protect
+  // against CSRF attacks, and that is to simply add another header. If
+  // HS2 is running with an CSRF filter enabled, then it will reject all
+  // requests that do not contain this. Thus, we add this in here on the
+  // client-side. This simple check prevents random other websites from
+  // redirecting a browser that has login credentials from making a
+  // request to HS2 on their behalf.
+
+  private static boolean injectHeader = true;
```

Comment:
why is it static?

Reply:
that variable is intended for testing purpose only. We cannot get the instance to make any change once it is created " httpClientBuilder.addInterceptorLast(new CsrfHttpRequestInterceptor());" in HiveConnection.java, similar to XSRF We should be injecting the header by default for all jdbc connections as we don't know whether filtering is enabled or not. Please let me know if anything else can be done